core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -86,16 +89,37 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
 
     import data.sparkSession.implicits._
 
-    val labels = data.map(r => r.getDouble(0) -> 1L)
-    val labelCounts = labels.groupBy(labels.columns(0)).sum(labels.columns(1)).persist()
-    val (resKeep, resDrop) = estimate(labelCounts)
+    val labels = data.map(r => r.getDouble(0))
+    val labelCounts = labels.groupBy(labels.columns(0)).count().persist()
+    val res = estimate(labelCounts)
     labelCounts.unpersist()
-    setLabels(resKeep, resDrop)
+    setLabels(res.labelsKept.toSet, res.labelsDropped.toSet, res.labelsDroppedTotal)
 
-    summary = Option(DataCutterSummary(labelsKept = getLabelsToKeep, labelsDropped = getLabelsToDrop))
+    summary = Option(DataCutterSummary(
+      labelsKept = getLabelsToKeep,
+      labelsDropped = getLabelsToDrop,
+      labelsDroppedTotal = getLabelsDroppedTotal))
     summary
   }
 
+
+  private def getLabelsFromMetadata(data: DataFrame): Array[String] =
+    Try {
+      val labelSF = data.schema.head
+      val labelColMetadata = labelSF.metadata
+      log.info(s"Raw label column metadata: $labelColMetadata")
+      labelColMetadata
+        .getMetadata("ml_attr")
```

Comment:
object `org.apache.spark.ml.attribute.AttributeKeys` has these consts defined. please use those if possible.

Reply:
considered it but it's a package-private class in `org.apache.spark.ml.attribute`

Reply:
you can reach it by creating an object in the same namespace. please do so, since you are accessing the same const values numerous times in this code.

Reply:
too pragmatic to my taste but I don't mind